src/BinaryConstraintOps.h

```diff
@@ -110,6 +110,7 @@ inline BinaryConstraintOp convertOverloadedConstraint(
         default:
             assert(false && "Invalid constraint conversion");
     }
+    __builtin_unreachable();
```

Comment:
What platforms and compilers is this available on? You could use an assert instead.

Reply:
This intrinsic is available for gcc/clang. Unfortunately, an assert will not notify the compiler that this path is infeasible. Even an abort() will not work. The idea is to avoid warnings in clang/mac os x.

Reply:
I noticed that __builtin_unreachable() is not available for GCC V4.3 and below. However, this would not work for us at all. Hence it should not be an issue.

Reply:
I don't think it's available for msvc, either. An assert instead of the intrinsic works on g++-9 and clang++-9. I don't know about msvc, or even icc, but at least it will compile.